packages/ember-data/lib/system/record_array_manager.js

```diff
@@ -261,7 +261,7 @@ export default Ember.Object.extend({
   willDestroy: function(){
     this._super();
 
-    forEach(flatten(values(this.filteredRecordArrays.values)), destroy);
+    forEach(flatten(values(this.filteredRecordArrays._values)), destroy);
```

Comment:
re-writing this to use the map's `forEach` sounds like the best solution to me. @kimroen do you have time to update this pr to do that?